src/components/Calculator.js

```diff
@@ -1,63 +1,51 @@
-      result: prevState.operator1.concat(target.textContent),
-    }));
-  }
-
-  btnOnClick({ target }) {
-    this.setState(() => ({
-      operator1: target.textContent,
-    }));
-  }
-
-  btnOnClickResult() {
-    this.setState((prevState) => ({
-      result: prevState.operator1 * 2,
-    }));
+  btnOnClick(buttonName) {
+    this.setState((prevState) => (calculate(prevState, buttonName)));
   }
 
   render() {
-    const { result } = this.state;
+    const { total, next, operation } = this.state;
     return (
       <div className="container">
         <div className="result">
-          {result}
+          { total || '' }
+          { operation || '' }
+          { next || '' }
         </div>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-zero">0</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-dot">.</button>
-        <button type="button" onClick={this.btnOnClickResult} className="btn-accent btn-equal">=</button>
-        <button type="button" onClick={this.btnOnClick} className="btn-light btn-clear">AC</button>
-        <button type="button" onClick={this.btnOnClick} className="btn-light btn-toogle-sign">+/-</button>
-        <button type="button" onClick={this.btnOnClick} className="btn-light btn-percentage">%</button>
-        <button type="button" onClick={this.btnOnClick} className="btn-accent btn-divider">/</button>
-        <button type="button" onClick={this.btnOnClick} className="btn-accent btn-multiply">*</button>
-        <button type="button" onClick={this.btnOnClick} className="btn-accent btn-substract">-</button>
-        <button type="button" onClick={this.btnOnClick} className="btn-accent btn-plus">+</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-one">1</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-two">2</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-three">3</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-four">4</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-five">5</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-six">6</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-seven">7</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-eight">8</button>
-        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-nine">9</button>
+        <Button typeBtn="0" btnOnClick={this.btnOnClick} btnClass="btn-light btn-zero" />
+        <Button typeBtn="." btnOnclick={this.btnOnClick} btnClass="btn-light btn-dot" />
+        <Button typeBtn="=" btnOnClick={this.btnOnClick} btnClass="btn-accent btn-equal" />
```

Comment:
Kindly fix the functionality of the dot(`.`) button. Currently, when I click on it nothing happens. I should be able to calculate decimal numbers.

When I try to add a decimal number this happens:
![image](https://user-images.githubusercontent.com/38088818/170547519-7abc422f-294b-4d7a-bd1b-f050d0907d76.png)